### PR TITLE
Changed machinelearning swagger for new AMLCompute type

### PR DESCRIPTION
**Pull request by msoni3**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by salameer** [comment]
Reassigining as Pari is OOF

**Comment by anuchandy** [comment]
@msoni3 please fix errors in examples: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/447946059

**Review by anuchandy — CHANGES_REQUESTED** [review]
Fix errors in examples: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/447946059

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 1]
```diff
@@ -0,0 +1,1956 @@
...
+          "description": "Password of admin account",
+          "type": "string"
+        },
+        "publicKeyData": {
+          "description": "Public key data",
+          "type": "string"
+        },
+        "privateKeyData": {
+          "description": "Private key data",
+          "type": "string"
```
Comment (anuchandy) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: SSH PrivateKey is the secret and should not be set in the resource, e.g. see VM Compute key model [here](https://github.com/Azure/azure-rest-api-specs/blob/bc2b1356222a0bda2b5d9fd7fbad7483bfe1ec08/specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json#L5613)

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 2]
```diff
@@ -0,0 +1,1956 @@
...
+                  "type": "string"
+                },
+                "sshPort": {
+                  "description": "Port open for ssh connections.",
+                  "type": "integer"
+                },
+                "address": {
+                  "description": "Public IP address of the virtual machine.",
+                  "type": "string"
+                },
```
Comment (anuchandy) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: what is the format of this ip address? ipv4 or fqdn?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 3]
```diff
@@ -0,0 +1,1956 @@
...
+          "title": "Low priority capable",
+          "description": "Specifies if the virtual machine size supports low priority VMs."
+        },
+        "premiumIO": {
+          "type": "boolean",
+          "title": "Premium IO supported",
+          "description": "Specifies if the virtual machine size supports premium IO."
+        }
+      },
+      "description": "Describes the properties of a VM size."
```
Comment (anuchandy) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: We can mark all these properties as readonly right?

**Review by anuchandy — CHANGES_REQUESTED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 4]
```diff
@@ -0,0 +1,1956 @@
...
+      "discriminator": "computeType",
+      "properties": {
+        "computeType": {
+          "description": "The type of compute",
+          "$ref": "#/definitions/ComputeType"
+        },
+        "computeLocation": {
+          "description": "Location for the underlying compute",
+          "type": "string"
+        },
```
[the remark this answers is not in the document]
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: computeLocation represents the location of the underlying compute, that can be a different location from the resource (the represented compute under the workspace)

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json (hunk 1 above)
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: that is used only for attach (user is bringing their own VM), and is used for the initial connection, we do not store that key, and it does not get set on the resource.
There is a different api (the listKeys one), that users can use to retrieve an ssh key to connect to the machine, and the returned sshKey will be different that the one provided by the user during attach.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json (hunk 2 above)
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: we accept both

**Comment by seth-krantzler** [comment]
Will this include swagger information for Datalake and Databricks? Compute UX has a workitem to add attach support for these types for GA I believe. I have sent an email to Heather Spetalnick and Salil to try and gain clarity on this as soon as possible.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 5]
```diff
@@ -0,0 +1,1956 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the request failed.",
+            "schema": {
+              "$ref": "#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "post": {
```
Comment (RyanBensonMSFT) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: What is the purpose for POST on the resourceId?

POST is reserved for Actions which are children of resources. If you need this API, create an action called something like:

POST /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/descriptiveActionName

**Review by ryansbenson — CHANGES_REQUESTED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 6]
```diff
@@ -0,0 +1,1956 @@
...
+              "description": "AML Compute properties",
+              "type": "object",
+              "properties": {
+                "vmSize": {
+                  "description": "Virtual Machine Size",
+                  "type": "string"
+                },
+                "vmPriority": {
+                  "description": "Virtual Machine priority",
+                  "type": "string"
```
[the remark this answers is not in the document]
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: sure we will make it enum 
{ LowPriority, Dedicated }

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json [hunk 7]
```diff
@@ -0,0 +1,1956 @@
...
+                  "description": "Cluster full qualified domain name",
+                  "type": "string"
+                },
+                "systemServices": {
+                  "type": "array",
+                  "description": "System services",
+                  "items": {
+                    "$ref": "#/definitions/SystemService"
+                  }
+                },
```
[the remark this answers is not in the document]
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: sure, we should be able to change it readonly ..

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json (hunk 3 above)
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: sure, we should be able to change it to readonly ...

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json (hunk 5 above)
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: this API got removed as this was notused ..

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json (hunk 3 above)
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json]: sure, we should be able to change it to readonly ...s

**Comment by msoni3** [comment]
All comments are addressed as per feedback, please take a look, thank you.

**Review by ryansbenson — APPROVED** [review]
Approved

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json [hunk 8]
```diff
@@ -0,0 +1,1987 @@
...
+        },
+        {
+          "type": "object",
+          "properties": {
+            "databricksAccessToken": {
+              "description": "access token for databricks account.",
+              "type": "string"
+            }
+          }
+
```
Comment (anuchandy) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: Should this be `AMLCompute`? In many examples it is `AMLCompute` not `AmlCompute`. We need to make sure we are using the word `AMLCompute` in both examples and swagger spec, Please cross check all examples

**Review by anuchandy — CHANGES_REQUESTED** [review]
There are few model validation errors (as i pointed earlier) please take a look at the comment and fix them.

Even after fixing issues based on the comments, you still will see an error related to `getAMLCompute.json` example, we can ignore it now as it is a possible bug in oav (validation tool) https://github.com/Azure/oav/issues/349

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json (hunk 8 above)
Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: it should be AmlCompute ..

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json [hunk 9]
```diff
@@ -537,7 +537,7 @@
         "description": "Gets compute definition by its name. Any secrets (storage keys, service credentials, etc) are not returned - use 'keys' nested resource to get them.",
         "x-ms-examples": {
           "Get a AKS Compute": { "$ref": "./examples/getAKSCompute.json" },
-          "Get a AML Compute": { "$ref": "./examples/getAMLCompute.json" }
+          "Get a AML Compute": { "$ref": "./examples/getAmlCompute.json" }
```
Comment (anuchandy) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: The example reference name (e.g. `./examples/getAmlCompute.json`) is case sensitive, in this case the actual file name is `getAMLCompute.json` (AML in caps).  This is why the CI is failing and throwing syntax error so you might want to rename the example file to match the reference name.

Comment (msoni3) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: done

**Comment by anuchandy** [comment]
@msoni3 thanks for fixing the examples. Is it ready merge from your side? if yes update the title by removing [Do not merge] prefix.

**Comment by anuchandy** [comment]
@msoni3 please confirm whether this spec is read to merge? which means following criteria is met:

1. The service is deployed to at atleast one production region.

if yes, remove the "Don't Merge" from the title then i can merge PR.

If service is not deployed and if you know an expected date of deployment then please provide the date.

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
Service is deployed to production and GA is next week. 

Synced with @salameer regarding SDK generation - @msoni3 as discussed, you can update the readme file when you want to the SDK to be generated. you can request for SDK releases via portal -
https://portal.azure-devex-tools.com/

**Review by anuchandy — APPROVED** [review]

**Pull request merged**